packages/tpetra/core/src/Tpetra_Details_createMirrorView.hpp

```diff
@@ -148,6 +148,7 @@ class CreateMirrorViewFromUnmanagedHostArray<ValueType, OutputDeviceType, true,
     else {
       // No need to initialize, if we're going to copy into it anyway.
       outView_nc = nc_output_view_type (view_alloc (std::string (label), WithoutInitializing), inSize);
+      // DEEP_COPY REVIEW - HOST-TO-DEVICE
```

Comment:
The createMirrorView file is a way to create host views of (potentially) device data or *not* do it depending on the underlying architecture (which is handles via template partial specialization).  This stuff gets used in tests frequently, but also is littered all over the core classes.

Not sure it is the cleanest way to do this, but this is what we've got.

Reply:
Given the way there are two sets of device types InputDeviceType and OutputDevice type, I'm not 100% sure the device to host annotations are correct here.

Reply:
I've been commenting things as device-to-device if they could be device-to-device but not necessarily always. For host-to-device to device-to-host I tried to do that only when host was specific but sometimes it was hard to tell. I think we're currently planning to make all these cases run same device execution space so it may not be critical I got the annotations always correct. I am using them as a way to break down the project into smaller PRs which may help when issues come up.